python-lib/dku_io_utils/utils.py

```diff
@@ -0,0 +1,111 @@
+import io
+import dill as pickle
+import pandas as pd
+import json
+import gzip
+from safe_logger import SafeLogger
+
+logger = SafeLogger("Forecast plugin")
+
+
+def read_from_folder(folder, path, object_type):
```

Comment:
After more investigation, there seems to be an undocumented way in GluonTS to handle model serialization/deserialization: https://github.com/awslabs/gluon-ts/issues/126
Now I am not sure this would work with our custom AutoArima model. We would need

I propose we postpone this to a later version. The current way we do things with pickle is the same as the Visual ML native feature, with the same security implications. There is a level of protection in UIF installations because pickle files written by user1 cannot be overwritten by another user.

What do you think?

Reply:
also one reason we didn't use this serialization/deserialization of GluonTS is the warning they put in their tutorial:
https://ts.gluon.ai/examples/extended_forecasting_tutorial/extended_tutorial.html#3.3-Saving/Loading-an-existing-model
`WARNING:root:Serializing RepresentableBlockPredictor instances does not save the prediction network structure in a backwards-compatible manner. Be careful not to use this method in production.`

Reply:
Ok to go with pickle for now.

Reply:
put this in the backlog: https://app.clubhouse.io/dataiku/story/59915/change-from-pickle-format-to-a-more-secure-one-to-save-models